draft-irtf-cfrg-hpke.md

```diff
@@ -298,56 +315,71 @@ operations, roles, and behaviors of HPKE:
 HPKE variants rely on the following primitives:
 
 * A Key Encapsulation Mechanism (KEM):
-  - GenerateKeyPair(): Generate a key pair (skX, pkX)
-  - DeriveKeyPair(ikm): Derive a key pair `(skX, pkX)` from the byte string `ikm`,
-    where `ikm` SHOULD have at least `Nsk` bytes of entropy (see
-    {{derive-key-pair}} for discussion)
-  - Serialize(pk): Produce a byte string of length `Npk` encoding the
-    public key `pk`
-  - Deserialize(enc): Parse the byte string `enc` of length `Npk` to recover a
-    public key (note: this function can raise an error upon `enc` deserialization failure)
-  - Encap(pk): Generate an ephemeral, fixed-length symmetric key (the KEM shared secret) and
-    a fixed-length encapsulation of that key that can be decapsulated
-    by the holder of the private key corresponding to pk
+  - GenerateKeyPair(): Generate a key pair `(skX, pkX)`.
+  - DeriveKeyPair(ikm): Derive a key pair `(skX, pkX)` from the byte
+    string `ikm`, where `ikm` SHOULD have at least `Nsk` bytes of entropy
+    (see {{derive-key-pair}} for discussion).
+  - SerializePublicKey(pkX): Produce a byte string of maximum length `Npk`
+    encoding the KEM public key `pkX`. A KEM MAY provide multiple
+    different equivalent encoding algorithms for public keys, with
+    different output lengths.
+  - DeserializePublicKey(pkXm): Parse the byte string `pkXm` of length `Npk`
+    to recover a KEM public key. This function can raise an error upon
+    `pkXm` deserialization failure.
+  - SerializeEncapsulatedKey(pkE): Produce a byte string of maximum length `Nenc`
```

Comment:
This change takes the complexity hit so that future documents don't need to. If we don't want this complexity, then I concur with the simplification above. (I don't think it's much complexity, so I'm OK adding it here.)

Reply:
I like the simpler approach. I was not happy about the clutter I introduced with the other (De)Serialize functions.

Reply:
Also, while it has been pointed out to us that there is a post-quantum KEM that provides different encodings for public keys, nobody told us about a KEM with different encodings for encapsulated keys, so this was just an idea that theoretically this could be the case. I did not have time yet to look at all the post-quantum KEM proposals.

Anyway, all this to say that I am fine with dropping this explicit definitions.